modules/afmongodb/afmongodb-legacy-grammar.c

```diff
@@ -0,0 +1,151 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "afmongodb-legacy-grammar.h"
+#include "afmongodb-legacy-private.h"
+
+gboolean
+afmongodb_dd_validate_socket_combination(LogDriver *d)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  if ((self->port != 0 || self->port != MONGO_CONN_LOCAL) && self->address != NULL)
+    return FALSE;
+  if (self->servers)
+    return FALSE;
+
+  return TRUE;
+}
+
+gboolean
+afmongodb_dd_validate_network_combination(LogDriver *d)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  if (self->port == MONGO_CONN_LOCAL && self->address != NULL)
+    return FALSE;
+
+  return TRUE;
+}
+
+void
+afmongodb_dd_set_user(LogDriver *d, const gchar *user)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using username() option is deprecated in mongodb driver, please use uri() instead");
+
+  g_free(self->user);
+  self->user = g_strdup(user);
+  self->is_legacy = TRUE;
+}
+
+void
+afmongodb_dd_set_password(LogDriver *d, const gchar *password)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using password() option is deprecated in mongodb driver, please use uri() instead");
+
+  g_free(self->password);
+  self->password = g_strdup(password);
+  self->is_legacy = TRUE;
+}
+
+void
+afmongodb_dd_set_host(LogDriver *d, const gchar *host)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using host() option is deprecated in mongodb driver, please use uri() instead");
+
+  g_free(self->address);
+  self->address = g_strdup(host);
+  self->is_legacy = TRUE;
+}
+
+void
+afmongodb_dd_set_port(LogDriver *d, gint port)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using port() option is deprecated in mongodb driver, please use uri() instead");
+
+  self->port = port;
+  self->is_legacy = TRUE;
+}
+
+void
+afmongodb_dd_set_servers(LogDriver *d, GList *servers)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using servers() option is deprecated in mongodb driver, please use uri() instead");
+
+  string_list_free(self->servers);
+  self->servers = servers;
+  self->is_legacy = TRUE;
+}
+
+void
+afmongodb_dd_set_path(LogDriver *d, const gchar *path)
+{
+  MongoDBDestDriver *self = (MongoDBDestDriver *)d;
+
+  msg_warning_once(
+      "WARNING: Using path() option is deprecated in mongodb driver, please use uri() instead");
+
+  g_free(self->address);
+  self->address = g_strdup(path);
+  self->port = MONGO_CONN_LOCAL;
```

Comment:
Is it a good idea to use address for both the UNIX domain path and the name of the server? It doesn't make it very easy to decipher validate_xxx_combination above.